pdns/pdns_recursor.cc

```diff
@@ -1760,12 +1762,16 @@ static void startDoResolve(void *p)
           dc->d_socket = -1;
         }
         else {
-          dc->d_tcpConnection->state=TCPConnection::BYTE0;
           Utility::gettimeofday(&g_now, 0); // needs to be updated
           struct timeval ttd = g_now;
-          ttd.tv_sec += g_tcpTimeout;
-
-          t_fdm->addReadFD(dc->d_socket, handleRunningTCPQuestion, dc->d_tcpConnection, &ttd);
+          dc->d_tcpConnection->d_requestsInFlight--;
+          if (dc->d_tcpConnection->d_requestsInFlight == TCPConnection::s_maxInFlight - 1) {
+            //cerr << "Reenabling... " << dc->d_tcpConnection->d_requestsInFlight << ' ' << dc->d_socket << endl;
+            ttd.tv_sec += g_tcpTimeout;
+            t_fdm->addReadFD(dc->d_socket, handleRunningTCPQuestion, dc->d_tcpConnection, &ttd);
+          } else {
+            t_fdm->setReadTTD(dc->d_socket, ttd,  g_tcpTimeout);
+          }
```

Comment:
It looks like we disable the polling on a timeout even if we have queries in-flight (good), but we could resume it later if `dc->d_tcpConnection->d_requestsInFlight == TCPConnection::s_maxInFlight - 1` which might not be what we want?